gitpod-io/gitpod
# Implement a 'Use Last Successful Prebuild' workspace creation mode

## Description


Implement a 'Use Last Successful Prebuild' workspace creation mode, and companion button in the "Prebuild In Progress" screen.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/12582

## How to test


1. Trigger a long-running prebuild, and let it complete successfully
2. Trigger a new long-running prebuild by pushing a new commit
3. Without waiting for the 2nd prebuild to complete, open the new commit in Gitpod
4. In the "Prebuild in Progress" screen, you should see a new button: "Use Last Successful Prebuild"
5. Clicking that button should give you a workspace based on the older successful prebuild (fast, no waiting)

## Release Notes

```release-note
Implement a 'Use Last Successful Prebuild' workspace creation mode
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`
